README.md

```diff
@@ -1,2 +1,161 @@
+
+A relay is needed in situations where nodes are behind NAT, reverse proxies, firewalls and/or simply don't support the same transports (e.g. go-libp2p vs. browser-libp2p). The circuit relay protocol exists to overcome those scenarios. Nodes with the `auto-relay` feature enabled can automatically bind themselves on a relay to listen for connections on their behalf.
+
+You can read more in its [SPEC](https://github.com/libp2p/specs/tree/master/relay).
+
+## Usage
+
+### Install
+
+```bash
+> npm install --global libp2p-relay-server
+```
+
+Now you can use the cli command `libp2p-relay-server` to spawn an out of the box libp2p relay server.
+
+### CLI
+
+After installing the relay server, you can use its binary. It accepts several arguments: `--peerId`, `--listenMultiaddrs`, `--announceMultiaddrs`, `--metricsMultiaddr`, `--disableMetrics` and `--disablePubsubDiscovery`.
+
+```sh
+libp2p-relay-server [--peerId <jsonFilePath>] [--listenMultiaddrs <ma> ... <ma>] [--announceMultiaddrs <ma> ... <ma>] [--metricsMultiaddr <ma>] [--disableMetrics] [--disablePubsubDiscovery]
+```
+
+The main configuration you should focus are the `PeerId` and the `Multiaddrs`, which are detailed next.
+
+#### PeerId
+
+You can create a [PeerId](https://github.com/libp2p/js-peer-id) via its [CLI](https://github.com/libp2p/js-peer-id#cli). 
+
+```sh
+libp2p-relay-server --peerId id.json
+```
+
+#### Multiaddrs
+
+You can specify the libp2p rendezvous server listen and announce multiaddrs. This server is configured with [libp2p-tcp](https://github.com/libp2p/js-libp2p-tcp) and [libp2p-websockets](https://github.com/libp2p/js-libp2p-websockets) and addresses with this transports should be used. It can always be modified via the API.
+
+```sh
+libp2p-relay-server --peerId id.json --listenMultiaddrs '/ip4/127.0.0.1/tcp/15002/ws' '/ip4/127.0.0.1/tcp/8000' --announceMultiaddrs '/dns4/test.io/tcp/443/wss/p2p/12D3KooWAuEpJKhCAfNcHycKcZCv9Qy69utLAJ3MobjKpsoKbrGA' '/dns6/test.io/tcp/443/wss/p2p/12D3KooWAuEpJKhCAfNcHycKcZCv9Qy69utLAJ3MobjKpsoKbrGA'
+```
+
+By default it listens on `/ip4/127.0.0.1/tcp/15003/ws` and has no announce multiaddrs specified.
+
+### Docker
+
+When running the relay server in Docker, you can configure the same parameters via environment variables, as follows:
+
+```sh
+PEER_ID='./id.json'
+LISTEN_MULTIADDRS='/ip4/127.0.0.1/tcp/15002/ws,/ip4/127.0.0.1/tcp/8000'
+ANNOUNCE_MULTIADDRS='/dns4/test.io/tcp/443/wss/p2p/12D3KooWAuEpJKhCAfNcHycKcZCv9Qy69utLAJ3MobjKpsoKbrGA,/dns6/test.io/tcp/443/wss/p2p/12D3KooWAuEpJKhCAfNcHycKcZCv9Qy69utLAJ3MobjKpsoKbrGA'
+```
+
+Please note that you should expose expose the used ports with the docker run command. The default ports used are `8003` for the metrics and `150003` for the websockets listener.
+
+Example:
+
+```sh
+docker build NAME -t libp2p-relay
+docker run -p 8003:8003 -p 15002:15002 -p 8000:8000 -e LISTEN_MULTIADDRS='/ip4/127.0.0.1/tcp/15002/ws,/ip4/127.0.0.1/tcp/8000' -d libp2p-relay
```

Comment:
It would be helpful to show how to include the peer id file here. Setting `PEER_ID='./id.json'` will fail because no file has been given to docker. This example is just going to create new peer ids.

Also, if we added announce addresses to the example `-e ANNOUNCE_MULTIADDRS='/dns4/localhost/tcp/8000,/dns4/localhost/tcp/15002/ws'`, people should be able to play around with this locally.